### PR TITLE
Fix for issue #834: DoCM runs GATK Haplotype Caller on entire genome

### PR DESCRIPTION
**Pull request by saimukund20**

### CODE REVIEW

**Comment by chrisamiller** [comment]
fix for https://github.com/genome/analysis-workflows/issues/834

File: definitions/tools/docm_gatk_haplotype_caller.cwl [hunk 1]
```diff
@@ -2,47 +2,59 @@
...
-    ["-gt_mode", "GENOTYPE_GIVEN_ALLELES",
-    "-o", { valueFrom: $(runtime.outdir)/docm_raw_variants.vcf }]
+      dockerPull: "broadinstitute/gatk:4.1.2.0"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'docm_haplotypeCaller.sh'
+        entry: |
+             set -o pipefail
+             set -o errexit
+
```
Comment (chrisamiller) [inline definitions/tools/docm_gatk_haplotype_caller.cwl]: To help the person who has to debug this someday (and that person might be you 😆 ), can you explicitly capture variables into names?   

reference=$1
vcf=$2   

(or whatever - I made those up)

Comment (saimukund20) [reply definitions/tools/docm_gatk_haplotype_caller.cwl]: Changed the file

**Comment by saimukund20** [comment]
Ill wait for another +1 before merging

**Pull request merged**